system-test/read-modify-write-row-interceptors.ts

```diff
@@ -0,0 +1,294 @@
+      );
+      // 2. Tell the metrics collector an attempt has been started.
+      metricsCollector.onOperationStart();
+      metricsCollector.onAttemptStart();
+      // 3. Make a unary call with gax options that include interceptors. The
+      // interceptors are built from a method that hooks them up to the
+      // metrics collector
+      const responseArray = await new Promise((resolve, reject) => {
+        bigtable.request(
+          {
+            client: 'BigtableClient',
+            method: 'readModifyWriteRow',
+            reqOpts: {
+              tableName: table.name,
+              rowKey: Buffer.from(ROW_KEY),
+              rules: [
+                {
+                  familyName: COLUMN_FAMILY,
+                  columnQualifier: Buffer.from(COLUMN),
+                  appendValue: Buffer.from('-wood'),
+                },
+              ],
+              appProfileId: undefined,
+            },
+            gaxOpts: withInterceptors({}, metricsCollector),
+          },
+          (err: ServiceError | null, resp?: any) => {
+            if (err) {
+              reject(err);
+            } else {
+              resolve(resp);
+            }
+          },
+        );
+      });
+      // 4. Tell the metrics collector the attempt is over
+      metricsCollector.onAttemptComplete(GrpcStatus.OK);
+      metricsCollector.onOperationComplete(GrpcStatus.OK);
+      // 5. Return results of method call to the user
+      return responseArray;
+    };
+
+    await (table as any).fakeReadModifyWriteRowMethod();
+
+    assert.strictEqual(testMetricsHandler.requestsHandled.length, 2);
+
+    const attemptCompleteData = testMetricsHandler.requestsHandled.find(
+      m => (m as {attemptLatency?: number}).attemptLatency !== undefined,
+    ) as OnAttemptCompleteData | undefined;
+    const operationCompleteData = testMetricsHandler.requestsHandled.find(
+      m => (m as {operationLatency?: number}).operationLatency !== undefined,
+    ) as OnOperationCompleteData | undefined;
+
+    assert.ok(attemptCompleteData, 'OnAttemptCompleteData should be present');
+    assert.ok(
+      operationCompleteData,
+      'OnOperationCompleteData should be present',
+    );
+    if (!attemptCompleteData || !operationCompleteData) {
+      throw new Error('Metrics data is missing'); // Should be caught by asserts above
+    }
+    assert.strictEqual(
+      attemptCompleteData.metricsCollectorData.method,
+      MethodName.READ_MODIFY_WRITE_ROW,
+    );
+    assert.strictEqual(attemptCompleteData.status, '0');
+    assert.strictEqual(
+      attemptCompleteData.metricsCollectorData.table,
+      TABLE_ID,
+    );
+    assert.strictEqual(
+      attemptCompleteData.metricsCollectorData.instanceId,
+      INSTANCE_ID,
+    );
+    assert.ok(attemptCompleteData.attemptLatency >= 0);
+    assert(attemptCompleteData.serverLatency);
+    assert.ok(attemptCompleteData.serverLatency >= 0);
```

Comment:
This is where we prove we are getting the server latency from the metadata headers.